extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/MultiStageQueryContext.java

```diff
@@ -204,6 +211,16 @@ public static int getRowsPerSegment(final QueryContext queryContext)
     );
   }
 
+  public static MSQSelectDestination getSelectDestination(final QueryContext queryContext)
+  {
+    return MSQSelectDestination.valueOf(
```

Comment:
What happens if this isn't a valid destination? If this errors out with a cryptic error message, we should present a user-friendly message and list out the possible options for the destination. (Presence of DURABLE_STORAGE/S3 as an option should only be if durable storage is configured)

Reply:
It would currently return an IAE with "value not found in enum", but the error message could be improved here. I think that's a good point, that we should make it more user friendly. This is true for all other enum query context parameters (workerAssignementStrategy, clusterStatisticsMergeMode etc). I'll make this change in a separate PR and include all other query context parameters as well. 

One thought, maybe DURABLE_STORAGE should be present always so that the user knows that it is an option and can configure durable storage if needed. If it is not enabled, we would throw an exception accordingly from wherever it is writing the results anyway. WDYT?

Reply:
Sure, makes sense